test/boost/sstable_datafile_test.cc

```diff
@@ -5701,15 +5701,20 @@ SEASTAR_TEST_CASE(sstable_run_based_compaction_test) {
             // check that sstable replacement follows token order
             BOOST_REQUIRE(*expected_sst == old_sstables.front()->generation());
             expected_sst++;
-            // check that previously released sstable was already closed
-            BOOST_REQUIRE(*closed_sstables_tracker == old_sstables.front()->generation());
+            // check that previously released sstables were already closed
+            if (old_sstables.front()->generation() % 4 == 0) {
+                // Due to performance reasons, sstables are not released immediately, but in batches.
+                // At the time of writing, mutation_reader_merger releases it's sstable references
+                // in batches of 4. That's why we only perform this check every 4th sstable. 
+                BOOST_REQUIRE(*closed_sstables_tracker == old_sstables.front()->generation());
+            }
```

Comment:
I wonder if that batching has negative impact on TWCS reads/LCS scans, since it could take a long time for a batch to accumulate. Of course for this series we won't touch it.